google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/TopicAdminSettings.java

```diff
@@ -106,131 +75,69 @@
+    return ((PublisherStubSettings) getStubSettings()).publishSettings();
   }
 
   /** Returns the object with the settings used for calls to getTopic. */
   public UnaryCallSettings<GetTopicRequest, Topic> getTopicSettings() {
-    return getTopicSettings;
+    return ((PublisherStubSettings) getStubSettings()).getTopicSettings();
   }
 
   /** Returns the object with the settings used for calls to listTopics. */
   public PagedCallSettings<ListTopicsRequest, ListTopicsResponse, ListTopicsPagedResponse>
       listTopicsSettings() {
-    return listTopicsSettings;
+    return ((PublisherStubSettings) getStubSettings()).listTopicsSettings();
   }
 
   /** Returns the object with the settings used for calls to listTopicSubscriptions. */
   public PagedCallSettings<
           ListTopicSubscriptionsRequest, ListTopicSubscriptionsResponse,
           ListTopicSubscriptionsPagedResponse>
       listTopicSubscriptionsSettings() {
-    return listTopicSubscriptionsSettings;
+    return ((PublisherStubSettings) getStubSettings()).listTopicSubscriptionsSettings();
   }
 
   /** Returns the object with the settings used for calls to deleteTopic. */
   public UnaryCallSettings<DeleteTopicRequest, Empty> deleteTopicSettings() {
-    return deleteTopicSettings;
+    return ((PublisherStubSettings) getStubSettings()).deleteTopicSettings();
   }
 
   /** Returns the object with the settings used for calls to setIamPolicy. */
   public UnaryCallSettings<SetIamPolicyRequest, Policy> setIamPolicySettings() {
-    return setIamPolicySettings;
+    return ((PublisherStubSettings) getStubSettings()).setIamPolicySettings();
   }
 
   /** Returns the object with the settings used for calls to getIamPolicy. */
   public UnaryCallSettings<GetIamPolicyRequest, Policy> getIamPolicySettings() {
-    return getIamPolicySettings;
+    return ((PublisherStubSettings) getStubSettings()).getIamPolicySettings();
   }
 
   /** Returns the object with the settings used for calls to testIamPermissions. */
   public UnaryCallSettings<TestIamPermissionsRequest, TestIamPermissionsResponse>
       testIamPermissionsSettings() {
-    return testIamPermissionsSettings;
+    return ((PublisherStubSettings) getStubSettings()).testIamPermissionsSettings();
   }
 
   @BetaApi("A restructuring of stub classes is planned, so this may break in the future")
   public PublisherStub createStub() throws IOException {
-    if (getTransportChannelProvider()
-        .getTransportName()
-        .equals(GrpcTransportChannel.getGrpcTransportName())) {
-      return GrpcPublisherStub.create(this);
-    } else {
-      throw new UnsupportedOperationException(
-          "Transport not supported: " + getTransportChannelProvider().getTransportName());
-    }
-  }
-
-  /** Returns a builder for the default ExecutorProvider for this service. */
-  public static InstantiatingExecutorProvider.Builder defaultExecutorProviderBuilder() {
```

Comment:
Can a TopicAdminSettings take in any StubSettings, or does it only take in a PublisherStubSettings? I'm not sure if these public static methods should remain in this class or the stub settings class.

Reply:
Only PublisherStubSettings. 

I think they should remain here - they are part of the surface of settings classes for GA clients.

Reply:
Ok, left the public static methods in TopicAdminSettings. However, deferred implementation of the methods to PublisherStubSettings, because the stub settings object is responsible for setting those default fields in its own builder.